src/engine.rs

```diff
@@ -69,9 +70,7 @@ pub(crate) async fn start(
     let transport = build_transport(local_keypair.clone()).await?;
     let behaviour = Behaviour::new(
         local_public_address,
-        &local_keypair,
-        10,
-        &config.kademlia_protocol_names, //TODO jonas
+        &local_keypair
```

Comment:
As mentioned previously, protocol name is no longer a constant, its a configurable field in Config, so we might have to  pass in the Config instance.